tests/models/test_hooks.py

```diff
@@ -497,6 +540,56 @@ def test_trainer_model_hook_system_test(tmpdir):
         'on_test_model_train',
         'teardown',
     ]
+    called = [c['name'] for c in called]
+    assert called == expected
+
+
+def test_trainer_model_hook_system_predict(tmpdir):
+    called = []
+    model = HookedModel(called)
+    callback = HookedCallback(called)
+    batches = 2
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        limit_predict_batches=batches,
+        progress_bar_refresh_rate=0,
+        callbacks=[callback],
+    )
+    assert called == [
+        dict(name='Callback.on_init_start', args=(trainer, )),
+        dict(name='Callback.on_init_end', args=(trainer, )),
+    ]
+    trainer.predict(model)
+    expected = [
+        dict(name='Callback.on_init_start', args=(trainer, )),
```

Comment:
Nit: {} instead of dict calls. [Ref](https://doughellmann.com/posts/the-performance-impact-of-using-dict-instead-of-in-cpython-2-7-2/)

Reply:
I also prefer dict literal over constructor, but unfortunately, yapf produces horrors when dict literals are used.

Can be changed when #7783 lands